pincer/middleware/interaction_create.py

```diff
@@ -145,8 +164,16 @@ async def interaction_handler(
     }
     params = {}
 
-    if interaction.data.options is not MISSING:
-        params = {opt.name: opt.value for opt in interaction.data.options}
+    options = interaction.data.options
+
+    if options and options[0].type in {1, 2}:
+        if options[0].type == 1:
+            options = options[0].options
+        else:
+            options = options[0].options[0].options
```

Comment:
what tf is this? xD

Tbh, remind me of tree like structure, where you can use recursion to get child info...?

Reply:
I think recursion makes it worse
```python
    # Recursion
    def get_options_from_command(options):
        if not options:
            return options
        if options[0].type == 1:
            return options[0].options
        if options[0].type == 2:
            return get_options_from_command(options[0].options)
        return options

    options = get_options_from_command(interaction.data.options)

    # Updated original version
    if options:
        if options[0].type == 1:
            options = options[0].options
        elif options[0].type == 2:
            options = options[0].options[0].options
```

Reply:
ah xD

Reply:
How would error handling work with that? 
I think im going to switch to recursion anyway because it makes what im trying to do very obvious.